src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs

```diff
@@ -360,6 +378,10 @@ private object CoerceText(object baseValue)
 			{
 				baseString = GetFirstLine(baseString);
 			}
+			else if (IsSkiaTextBox)
+			{
+				baseString = baseString.Replace("\r\n", "\r").Replace("\n", "\r");
```

Comment:
This is somewhat strange. Why are we replacing new lines with `\r`?

Reply:
That's the behaviour on WinUI, `\n`s aren't a thing, only `\r`s are allowed, unlike `TextBlock` which is okay with whatever you give it.

Reply:
I'd also use `ReplaceLineEndings` as it would be more efficient

```suggestion
				baseString = baseString.ReplaceLineEndings("\r");
```

Reply:
I remember ReplaceLineEndings failing on some older targets, so I've replaced it with what we have right now.

Reply:
As for why the change is only specific to SkiaTextBox, no specific reason. I've kept all the changes specific to skia TextBox so that we don't change the behaviour elsewhere (specifically the gtk/wpf textboxes).

Reply:
I think it shouldn't fail. We are net7.0+. But we can look into that in a separate PR to get this one merged faster.